### ISSUE TITLE
Schema update treatment protocol - 2022-02-08

### ISSUE THREAD

**Issue by ESapenaVentura** [ebi-ait/hca-ebi-wrangler-central#635]
**What are the next schema updates?**

- ~https://github.com/HumanCellAtlas/metadata-schema/issues/1441~
- [Treatment protocol](https://github.com/HumanCellAtlas/metadata-schema/issues/1428)  - PR [here](https://github.com/HumanCellAtlas/metadata-schema/pull/1446)
- new PR: HumanCellAtlas/schema-test-data#23

@ESapenaVentura is the wrangler responsible to create the schema PRs

**What schemas will they affect?**

- `type/protocol/imaging/imaging_protocol.json`

**Why are these changes needed?**

This change will facilitate the addition of imaging data

**Type of change and time frame**

Update number 1 constitutes a major change.

We are expecting to push both updates before 22nd February 2022

Current Actions:
- [ ] Update schema test data
- [ ] update geo to HCA template ebi-ait/geo_to_hca#37

**Comment by ESapenaVentura** [comment]
@Wkt8 Please send this ticket alongside the PR(s) https://github.com/HumanCellAtlas/metadata-schema/pull/1442 when there is clearance on the other update (e.g. after discussion with UCSC)

**Comment by ipediez** [comment]
waiting from input from other partners on the PRs

**Comment by Wkt8** [comment]
We will need to make test-data for the treatment_protocol metadata schema change.

**Comment by ami-day** [comment]
I have a spreadsheet that we can use as a test  - it is a standard spreadsheet and I will add a treatment protocol tab. Would that work as the test-data?

**Comment by ESapenaVentura** [comment]
@ami-day to speak with the ontology team about ontologies we could use on the treatment protocol

For 1428 we don't need test data, @Wkt8  to link to imaging schema-test-data PR and state that there won't be graph changes from that

**Comment by Wkt8** [comment]
Have linked to the imaging schema-test-data PR on the imaging_protocol PR.

**Comment by idazucchi** [comment]
@ESapenaVentura will merge the imaging PR

**Comment by ipediez** [comment]
HumanCellAtlas/metadata-schema#1441 is in master now

**Comment by ami-day** [comment]
We have decided with Paola on 'Assay" for the treatment protocol method ontology: https://www.ebi.ac.uk/ols/ontologies/efo/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FOBI_0000070&viewMode=All&siblings=false

**Comment by ami-day** [comment]
Updated the pull request to include the ontology: https://github.com/HumanCellAtlas/metadata-schema/pull/1446

**Comment by ipediez** [comment]
@ami-day what's the update on this?

**Comment by gabsie** [comment]
ping the reviewers on which this promotion depends on. @ami-day

**Comment by ofanobilbao** [comment]
It would probably require a discussion among the wranglers. Paris and Ami have to be there. Waiting now

**Comment by yusra-haider** [comment]
@ESapenaVentura , @Wkt8  and @ami-day to have a look

**Comment by gabsie** [comment]
@ami-day  - let us know if there's progress here.

**Comment by ami-day** [comment]
> @ami-day - let us know if there's progress here.

Hi @gabsie I have asked the ontology team to import the GO ontology to HCAO, I think we should wait to hear that that's done ok before re-review and approval of the this pull request

**Comment by ofanobilbao** [comment]
Pending ontologies and re-approvals

**Comment by gabsie** [comment]
Re-connect with the Ontology team to see if this has happened. Get the proper people tagged in order to get the approvals. @ami-day

**Comment by ESapenaVentura** [comment]
https://github.com/HumanCellAtlas/ontology/issues/111

**Comment by Wkt8** [comment]
Still waiting on discussion with zoe.

**Comment by ofanobilbao** [comment]
Zoe was meant to come back on 1st July

**Comment by gabsie** [comment]
I think there was another comment here, that the next HCAO release will be in August

**Comment by ami-day** [comment]
The pull request has been re-reviewed and merged into staging.

**Comment by ami-day** [comment]
Working on the post merging into staging release process

**Comment by amnonkhen** [comment]
@ami-day to sort github access with @amnonkhen

**Comment by ami-day** [comment]
Created PR for Release, Merge from Staging -> Master
https://github.com/HumanCellAtlas/metadata-schema/pull/1481

**Comment by ami-day** [comment]
> @ami-day to sort github access with @amnonkhen

done

**Comment by ami-day** [comment]
Updated the geo_to_hca spreadsheet template to reflect these schema updates. The template needs testing in ingest (e.g. if it validates or not for a test dataset): https://github.com/ebi-ait/geo_to_hca/pull/37

**Comment by ami-day** [comment]
Test data has been created and uploaded to staging: https://staging.contribute.data.humancellatlas.org/submissions/detail?uuid=7f33ae0e-656b-4987-83a8-d7e7cd0bb63f&project=14746c66-f605-4eb1-af65-f5cbf6658f82 It is not accepting the target pathway ontology term

**Comment by ESapenaVentura** [comment]
Steps needed to be done to correct the ontology issue:
- [ ] Delete the schema: target_pathway_ontology (https://schema.staging.data.humancellatlas.org/module/ontology/1.0.0/target_pathway_ontology) from the bucket so that any changes can be picked up by the publisher
- [ ] Similar to [the previous correction](https://github.com/HumanCellAtlas/metadata-schema/pull/1480) in staging, create a small PR to correct target_pathway_ontology `ontology.graph_restrictions.ontologies` field to `["obo:hcao"]`
- [ ] push that into staging (without bumping versions etc)
- [ ] Make sure that the new version is published by checking the messages in #hca-schema-pub-annouce (Slack)
- [ ] Create a new submission (you can now delete the old one, I'm done testing) to create the test data

**Comment by gabsie** [comment]
Waiting for @ESapenaVentura to advise on some of the above steps.

**Comment by ESapenaVentura** [comment]
Step 1 is done. 

@ami-day do you understand how to do the rest of the steps? I can help you out when you create the PR / to create the PR if you don't know how to

**Comment by ESapenaVentura** [comment]
(Moved the ticket to in progress)

**Comment by gabsie** [comment]
Waiting for the ontology for data curator block.

**Comment by ESapenaVentura** [comment]
With #882 fix we should be able to generate the test data now

**Comment by gabsie** [comment]
Just to say, we have removed the priority for the experimental design test data so now the treatment protocol test data generation has taken over priority.
@ami-day has to generate the test data again, sync the files, export the data.
Tag the appropriate reviewers once ready (as per DCP2 SOPs)

**Comment by ami-day** [comment]
The test data seems to be stuck in exporting status in ingest staging: https://staging.contribute.data.humancellatlas.org/projects/detail?uuid=14746c66-f605-4eb1-af65-f5cbf6658f82&tab=upload

**Comment by MightyAx** [comment]
I don't know why this wasn't caught by validation but some process links don't match the schema because:
ValidationError: **`treatment_protocol` is not one of:**

- analysis_protocol
- aggregate_generation_protocol
- collection_protocol
- differentiation_protocol
- dissociation_protocol
- enrichment_protocol
- ipsc_induction_protocol
- imaging_preparation_protocol
- imaging_protocol
- library_preparation_protocol
- sequencing_protocol

```
ERROR - submission_uuid:27c552fc-bb6e-48f2-8223-dcd6eb57f2fa - export_job_id:6315dea206f1711fccbe5c39 -
Rejecting export experiment:
  ExperimentMessage( process_id='63037aef06f1711fccbe1cb6', process_uuid='1d8bdecb-a99d-4197-a464-a4e016e6ac94', submission_uuid='27c552fc-bb6e-48f2-8223-dcd6eb57f2fa', experiment_index=3, total=5, job_id='6315dea206f1711fccbe5c39')
due to error: (
  problem validating document {
    'links': [
      {'link_type': 'process_link', 'process_id': '1d8bdecb-a99d-4197-a464-a4e016e6ac94', 'process_type': 'process', 'inputs': [{'input_type': 'cell_suspension', 'input_id': '6b53df23-fafa-481f-a5d4-adf62e0f5571'}], 'outputs': [{'output_type': 'analysis_file', 'output_id': '8ea2808c-f6c8-4beb-ba1f-6993db9c0500'}], 'protocols': [{'protocol_type': 'analysis_protocol', 'protocol_id': '516eda77-e6b0-49f6-a9f4-5430c4ede28d'}, {'protocol_type': 'library_preparation_protocol', 'protocol_id': '2b8ce790-bb70-4777-8879-64fba10cf9d4'}]},
      {'link_type': 'process_link', 'process_id': '698dd4d0-ccea-452d-8180-dcaace21c238', 'process_type': 'process', 'inputs': [{'input_type': 'specimen_from_organism', 'input_id': '91759d1d-934e-42eb-8f95-58146c8b0f06'}], 'outputs': [{'output_type': 'cell_suspension', 'output_id': '6b53df23-fafa-481f-a5d4-adf62e0f5571'}], 'protocols': [{'protocol_type': 'enrichment_protocol', 'protocol_id': '519a9307-5e65-402a-8b9f-77178fd86377'}, {'protocol_type': 'treatment_protocol', 'protocol_id': 'f7bea781-ef8b-4b2c-9dad-b9a439780d11'}]},
      {'link_type': 'process_link', 'process_id': '1edf5525-1b6d-4b4d-bf51-7e7a5b7ca346', 'process_type': 'process', 'inputs': [{'input_type': 'donor_organism', 'input_id': 'b22a9e3f-6bf8-4152-bfb2-74a4e8f97e5c'}], 'outputs': [{'output_type': 'specimen_from_organism', 'output_id': '91759d1d-934e-42eb-8f95-58146c8b0f06'}], 'protocols': [{'protocol_type': 'collection_protocol', 'protocol_id': '2d21b68a-c9fa-4aa9-93d0-a6ebeed8dbdc'}]}
    ],
    'schema_type': 'links', 'describedBy': 'https://schema.staging.data.humancellatlas.org/system/3.0.0/links', 'schema_version': '3.0.0'
  }
  with schema: https://schema.staging.data.humancellatlas.org/system/3.0.0/links
  ValidationError: 'treatment_protocol' is not one of ['analysis_protocol', 'aggregate_generation_protocol', 'collection_protocol', 'differentiation_protocol', 'dissociation_protocol', 'enrichment_protocol', 'ipsc_induction_protocol', 'imaging_preparation_protocol', 'imaging_protocol', 'library_preparation_protocol', 'sequencing_protocol']
)

ERROR - submission_uuid:27c552fc-bb6e-48f2-8223-dcd6eb57f2fa - export_job_id:6315dea206f1711fccbe5c39 -
Rejecting export experiment:
  ExperimentMessage(process_id='63037aef06f1711fccbe1cb5', process_uuid='bed9e199-b465-49c1-b754-493d487d585e', submission_uuid='27c552fc-bb6e-48f2-8223-dcd6eb57f2fa', experiment_index=2, total=5, job_id='6315dea206f1711fccbe5c39')
due to error: (
  problem validating document {
    'links': [
      {'link_type': 'process_link', 'process_id': 'bed9e199-b465-49c1-b754-493d487d585e', 'process_type': 'process', 'inputs': [{'input_type': 'cell_suspension', 'input_id': 'b01e8852-1d52-41fa-bb58-c4ff13c31d5c'}], 'outputs': [{'output_type': 'analysis_file', 'output_id': '08c27b2c-9fe6-4a3b-ad9f-7d138555e03e'}], 'protocols': [{'protocol_type': 'analysis_protocol', 'protocol_id': '516eda77-e6b0-49f6-a9f4-5430c4ede28d'}, {'protocol_type': 'library_preparation_protocol', 'protocol_id': 'e60cd4c5-f38b-4d57-9a71-2bdf8bad1cc4'}]},
      {'link_type': 'process_link', 'process_id': '9e0e5e82-288b-431e-bec5-a27ff243aef0', 'process_type': 'process', 'inputs': [{'input_type': 'specimen_from_organism', 'input_id': '45c642e0-dbb3-4525-9838-2948a051c612'}], 'outputs': [{'output_type': 'cell_suspension', 'output_id': 'b01e8852-1d52-41fa-bb58-c4ff13c31d5c'}], 'protocols': [{'protocol_type': 'enrichment_protocol', 'protocol_id': '045172b2-5784-41a6-bf95-9853b1afc3e6'}, {'protocol_type': 'enrichment_protocol', 'protocol_id': 'b8b00bc3-d905-4dcb-aa01-40ab156eeb25'}, {'protocol_type': 'treatment_protocol', 'protocol_id': 'f7bea781-ef8b-4b2c-9dad-b9a439780d11'}]},
      {'link_type': 'process_link', 'process_id': '93545bda-737c-4333-a4a4-674ceb822e0d', 'process_type': 'process', 'inputs': [{'input_type': 'donor_organism', 'input_id': '7b599856-4104-415f-8f10-22690afb713f'}], 'outputs': [{'output_type': 'specimen_from_organism', 'output_id': '45c642e0-dbb3-4525-9838-2948a051c612'}], 'protocols': [{'protocol_type': 'collection_protocol', 'protocol_id': '15efb0a0-62f5-4ae7-835f-71b034d82687'}]}
    ],
    'schema_type': 'links', 'describedBy': 'https://schema.staging.data.humancellatlas.org/system/3.0.0/links', 'schema_version': '3.0.0'
  }
  with schema: https://schema.staging.data.humancellatlas.org/system/3.0.0/links
  ValidationError: 'treatment_protocol' is not one of ['analysis_protocol', 'aggregate_generation_protocol', 'collection_protocol', 'differentiation_protocol', 'dissociation_protocol', 'enrichment_protocol', 'ipsc_induction_protocol', 'imaging_preparation_protocol', 'imaging_protocol', 'library_preparation_protocol', 'sequencing_protocol']
)
```

**Comment by ESapenaVentura** [comment]
This issue should be unblocked now

**Comment by ami-day** [comment]
Re-submitted the test data, it is is exporting state: https://staging.contribute.data.humancellatlas.org/submissions/detail?uuid=eb3c4515-2109-4565-a641-3337b93571f4&project=92f00498-03d7-4dcc-b87d-6df5b18ac4d8

**Comment by ami-day** [comment]
Created new pull request: https://github.com/HumanCellAtlas/schema-test-data/pull/23

**Comment by gabsie** [comment]
@ami-day to nudge Dave and Nate, according to the SOP rules.

**Comment by ESapenaVentura** [comment]
Test data has been merged - however, the last step in the SOP is still missing, which is to ensure that the spreadsheet is up to date to the latest spreadsheet used

@ami-day please update the spreadsheet in the test-data repo

**Comment by gabsie** [comment]
A new ticket needs creating in the GEO-to-HCA repository with all the new updated columns with latest metadata schema introduction. @ESapenaVentura

**Comment by gabsie** [comment]
A new ticket needs creating in the GEO-to-HCA repository with all the new updated columns with latest metadata schema introduction. @ESapenaVentura

**Comment by MightyAx** [comment]
Actions:

- [ ] Update schema test data
- [ ] update geo to HCA template

**Comment by ESapenaVentura** [comment]
@idazucchi to review PR in schema-test-data repo

**Comment by idazucchi** [comment]
I think my review doesn't count towards the 3 mandatory reviews because I don't have write access to the repo - can some one add me?

**Comment by MightyAx** [comment]
I think I resolved the write access directly,
but I also invited Ida to: [dcp2-pr-reviewers](https://github.com/orgs/HumanCellAtlas/teams/dcp2-pr-reviewers/members)
which might be the more "correct" fix

**Comment by ESapenaVentura** [comment]
the schema test data is fine

It just needs one more review from a codeowner (@amnonkhen?)

**Comment by ami-day** [comment]
Done.

**Comment by idazucchi** [comment]
I've opened a [PR](https://github.com/HumanCellAtlas/schema-test-data/pull/29) for the test data SOP to address some of the issues we came across while working on this ticket
